src/main/java/org/embulk/gradle/runset/InstallEmbulkRunSet.java

```diff
@@ -0,0 +1,199 @@
+import org.gradle.api.artifacts.component.ComponentIdentifier;
+import org.gradle.api.artifacts.component.ModuleComponentIdentifier;
+import org.gradle.api.artifacts.component.ProjectComponentIdentifier;
+import org.gradle.api.artifacts.result.ArtifactResolutionResult;
+import org.gradle.api.artifacts.result.ArtifactResult;
+import org.gradle.api.artifacts.result.ComponentArtifactsResult;
+import org.gradle.api.artifacts.result.ResolvedArtifactResult;
+import org.gradle.api.file.DuplicatesStrategy;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.tasks.Copy;
+import org.gradle.maven.MavenModule;
+import org.gradle.maven.MavenPomArtifact;
+
+/**
+ * A Gradle Task to set up an environment for running Embulk.
+ */
+public class InstallEmbulkRunSet extends Copy {
+    public InstallEmbulkRunSet() {
+        super();
+
+        this.project = this.getProject();
+        this.logger = this.project.getLogger();
+
+        final ObjectFactory objectFactory = this.project.getObjects();
+    }
+
+    /**
+     * Adds a Maven artifact to be installed.
+     *
+     * <p>It tries to simulate Gradle's dependency notations, but it is yet far from perfect.
+     *
+     * @see <a href="https://github.com/gradle/gradle/blob/v8.4.0/subprojects/dependency-management/src/main/java/org/gradle/api/internal/notations/DependencyNotationParser.java#L49-L86">org.gradle.api.internal.notations.DependencyNotationParser#create</a>
+     */
+    public void artifact(final Object dependencyNotation) {
+        final Dependency dependency;
+        if (dependencyNotation instanceof CharSequence) {
+            dependency = this.dependencyFromCharSequence((CharSequence) dependencyNotation);
+        } else if (dependencyNotation instanceof Map) {
+            dependency = this.dependencyFromMap((Map) dependencyNotation);
+        } else {
+            throw new IllegalDependencyNotation("Supplied module notation is invalid.");
+        }
+
+        // Constructing an independent (detached) Configuration so that its dependencies are not affected by other plugins.
+        final Configuration configuration = this.project.getConfigurations().detachedConfiguration(dependency);
+
+        final ResolvableDependencies resolvavbleDependencies = configuration.getIncoming();
+        final ArtifactCollection artifactCollection = resolvavbleDependencies.getArtifacts();
+
+        // Getting the JAR files and component IDs.
+        final ArrayList<ComponentIdentifier> componentIds = new ArrayList<>();
+        for (final ResolvedArtifactResult resolvedArtifactResult : artifactCollection.getArtifacts()) {
+            componentIds.add(resolvedArtifactResult.getId().getComponentIdentifier());
+            this.fromArtifact(resolvedArtifactResult, "jar");
+        }
+
+        // Getting the POM files.
+        final ArtifactResolutionResult artifactResolutionResult = this.project.getDependencies()
```

Comment:
IDE says `Unchecked generics array creation for varargs parameter ` (Just in case comment).

Reply:
Yeah, this was unavoidable maybe because of the design problem of Gradle API. :(